### PR TITLE
Unquarantine passing tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
Tests have been passing for 30+ days.

### CODE REVIEW

File: src/ProjectTemplates/test/MvcTemplateTest.cs [hunk 1]
```diff
@@ -108,7 +108,6 @@ private async Task MvcTemplateCore(string languageOverride)
         [InlineData(true)]
         [InlineData(false)]
         [SkipOnHelix("cert failure", Queues = "OSX.1014.Amd64;OSX.1014.Amd64.Open")]
-        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/19716")]
```
Comment (BrennanConroy) [inline src/ProjectTemplates/test/MvcTemplateTest.cs:111]: Linked issue has been closed with a fix that we believe was affecting this test.

File: src/ProjectTemplates/test/WebApiTemplateTest.cs [hunk 2]
```diff
@@ -28,7 +28,6 @@ public WebApiTemplateTest(ProjectFactoryFixture factoryFixture, ITestOutputHelpe
 
         [ConditionalFact]
         [SkipOnHelix("Cert failures", Queues = "OSX.1014.Amd64;OSX.1014.Amd64.Open")]
-        [QuarantinedTest]
```
Comment (BrennanConroy) [inline src/ProjectTemplates/test/WebApiTemplateTest.cs:31]: No issue or logs associated with the failure that caused this to be quarantined are available.
It was quarantined 1.5 months ago and has had no failures in at least 30 days.

Propose un-quarantining and monitoring it for future failures.

File: src/ProjectTemplates/test/WorkerTemplateTest.cs [hunk 3]
```diff
@@ -22,7 +22,6 @@ public WorkerTemplateTest(ProjectFactoryFixture projectFactory, ITestOutputHelpe
         public ITestOutputHelper Output { get; }
 
         [Fact]
-        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/19716")]
```
Comment (BrennanConroy) [inline src/ProjectTemplates/test/WorkerTemplateTest.cs:25]: Linked issue has been closed with a fix that we believe was affecting this test.

File: src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs [hunk 4]
```diff
@@ -109,7 +109,6 @@ public static IEnumerable<object[]> LargeUploadData
         }
         [Theory]
         [MemberData(nameof(LargeUploadData))]
-        [QuarantinedTest]
```
Comment (BrennanConroy) [inline src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:112]: This one I'm not sure about.
I don't know if any work was done to fix this test, will leave to experts to decide.

Failure log in quarantine PR shows:
```
System.IO.IOException : Unable to read data from the transport connection: Broken pipe.
---- System.Net.Sockets.SocketException : Broken pipe
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.MaxRequestBufferSizeTests.<>c__DisplayClass4_0.<<LargeUpload>b__0>d.MoveNext() in /_/src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:line 141
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.MaxRequestBufferSizeTests.LargeUpload(Nullable`1 maxRequestBufferSize, Boolean connectionAdapter, Boolean expectPause) in /_/src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:line 190
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.SendAsyncForNetworkStream(Socket socket, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.SendAsyncForNetworkStream(ReadOnlyMemory`1 buffer, SocketFlags socketFlags, CancellationToken cancellationToken)
   at System.Net.Sockets.NetworkStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 size, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.MaxRequestBufferSizeTests.<>c__DisplayClass4_0.<<LargeUpload>b__0>d.MoveNext() in /_/src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:line 141
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.MaxRequestBufferSizeTests.<>c__DisplayClass4_0.<LargeUpload>b__0()
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.MaxRequestBufferSizeTests.LargeUpload(Nullable`1 maxRequestBufferSize, Boolean connectionAdapter, Boolean expectPause) in /_/src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:line 150
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
```

File: src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs [hunk 5]
```diff
@@ -1416,7 +1416,6 @@ public async Task Settings_MaxHeaderListSize_Client(string scheme)
         // Settings_InitialWindowSize_Lower_Client - Not configurable.
 
         [Theory]
-        [QuarantinedTest]
```
Comment (BrennanConroy) [inline src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs:1419]: No failure or issue for this one, propose un-quarantining and monitoring for future failure.

File: src/Tools/dotnet-user-secrets/test/SecretManagerTests.cs [hunk 6]
```diff
@@ -164,7 +164,6 @@ public void SetSecret_Update_Existing_Secret()
         }
 
         [Fact]
-        [QuarantinedTest]
```
Comment (BrennanConroy) [inline src/Tools/dotnet-user-secrets/test/SecretManagerTests.cs:167]: Failure when quarantined was `malloc_consolidate(): invalid chunk size`, that would be an issue in Runtime, but no other info was captured.

Propose un-quarantining, and if it fails again seeing if we captured a dump or figuring out if we need to add dump capturing.

**Review by Tratcher — APPROVED** [review]
@halter73 thoughts on LargeUpload?

File: src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs (hunk 4 above)
Comment (halter73) [reply src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:112]: This has historically been flaky, and no work that I know of has been done to fix it. I would just keep this one quarantined.

**Review by halter73 — APPROVED** [review]

**Review by Pilchie — APPROVED** [review]

**Pull request merged**